### PR TITLE
jump back and forth from summary page

### PR DESCRIPTION
**Pull request by mtamas5**
Modified user journey.
After pushing the change button on summary page the user can jump back to summary again.

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of nhsuk/register-with-a-gp-beta-web succeeded (http://register-with-a-gp-beta-web-pr-240.dev.beta.nhschoices.net)

File: src/server/plugins/register-form/steps/index.js [hunk 1]
```diff
@@ -82,4 +82,29 @@ const steps = [
   summaryStep
 ];
 
-export default steps;
+export const stepDependency = {
+  'nhs-number': [
+    'nhs-number-details'
```
Comment (erdem) [inline src/server/plugins/register-form/steps/index.js]: This dependency step is removed, this line must be removed.

File: src/server/plugins/register-form/steps/index.js [hunk 2]
```diff
@@ -82,4 +82,29 @@ const steps = [
...
+  ],
+  'home-address-lookup': [
+    'home-address-manual',
+    'home-address-lookup-choose'
+  ],
+  'previously-armed': [
+    'armed-staff-number',
```
Comment (erdem) [inline src/server/plugins/register-form/steps/index.js]: This dependency step is removed, this line must be removed.

File: src/__tests__/plugins/register-form/common.js [hunk 3]
```diff
@@ -87,6 +87,10 @@ describe('Get next step', () => {
     expect(checkStepCompletedBefore(requestedStepKey, latestUncompletedStep)).toEqual(true);
   });
 
+  it('should call getSlugByKey and return right practice', () => {
+    expect(getSlugByKey('nhsNumber')).toEqual('nhs-number');
+  });
+
```
Comment (erdem) [inline src/__tests__/plugins/register-form/common.js]: We need to implement new unit tests for whole jump back flow.

**Pull request merged**